src/main.rs

```diff
@@ -11,16 +11,19 @@ mod proto_helpers;
 mod replication;
 mod storage;
 mod types;
+mod minitrace_grpc;
 
 use crate::cache::Cache;
 use crate::config::Config;
+use crate::minitrace_grpc::{MinitraceGrpcMiddlewareLayer, MinitraceSpans, report_datadog_traces};
 use crate::object::ObjectGrpc;
 use crate::public_key::PublicKeyGrpc;
 use crate::mailbox::MailboxGrpc;
 use crate::types::{OsError, Result};
 use crate::storage::FileSystem;
 use crate::replication::{reap_unknown_keys, replicate, ReplicationState};
 
+use std::sync::mpsc::channel;
```

Comment:
Since this is being used in an async context, I think we need to use the tokio mpsc channel. I believe this channel blocks based on how the channel is configured and which calls are being made.